### PR TITLE
EVG-13069 Add logging for build baron related tickets bug

### PR DESCRIPTION
**Pull request by malikchaya2**

### CODE REVIEW

File: graphql/ui_plugin_build_baron.go [hunk 1]
```diff
@@ -127,6 +128,9 @@ func GetSearchReturnInfo(taskId string, exec string) (*thirdparty.SearchReturnIn
 
 	if !ok {
 		projectNotFoundError = true
+		grip.Debug(message.Fields{
+			"message": "Chaya, ui_plugin_build_baron.go, 132",
```
Comment (ybrill) [inline graphql/ui_plugin_build_baron.go:132]: I've found it useful to log each field I want to search on separately. Like I'd have "ticket", "function", and "line" fields. It's made searching Splunk easier: I can find all the logging for a particular ticket or for a particular function, for example.

File: graphql/ui_plugin_build_baron.go [hunk 2]
```diff
@@ -139,9 +143,19 @@ func GetSearchReturnInfo(taskId string, exec string) (*thirdparty.SearchReturnIn
...
+		grip.Debug(message.Fields{
+			"message": "Chaya, graphql/ui_plugin_build_baron.go, 147",
+			"err":     err,
+			"tickets": tickets,
+			"source":  source,
+		})
 		return nil, projectNotFoundError, errors.New(fmt.Sprintf("Error searching for tickets: %s", err.Error()))
 	}
 	jql := t.GetJQL(bbProj.TicketSearchProjects)
+	grip.Debug(message.Fields{
```
Comment (ybrill) [inline graphql/ui_plugin_build_baron.go:155]: I fear logging every query (not only the errors) might be a lot.
If you agree, maybe line 154 could be moved before 146 and the jql could only be logged in the error case, together with the previous log message.

File: graphql/ui_plugin_build_baron.go [hunk 3]
```diff
@@ -139,9 +143,19 @@ func GetSearchReturnInfo(taskId string, exec string) (*thirdparty.SearchReturnIn
 
 	tickets, source, err = multiSource.Suggest(t)
 	if err != nil {
+		grip.Debug(message.Fields{
+			"message": "Chaya, graphql/ui_plugin_build_baron.go, 147",
+			"err":     err,
+			"tickets": tickets,
```
Comment (ybrill) [inline graphql/ui_plugin_build_baron.go:149]: I fear logging the entire response might be a lot. (I got a 215kB response for [one query](https://jira.mongodb.org/rest/api/latest/search?jql=%28project%20in%20%28BF%2C%20WT%2C%20SERVER%29%29%20and%20%28%20text~%5C%22db_repl_test%5C%22%20%29%20order%20by%20updatedDate%20desc&startAt=0&maxResults=-1))

Maybe we could limit it to just the beginning (150 runes?) just to get a sense of what it looks like?

File: service/ui_plugin_build_baron.go [hunk 4]
```diff
@@ -120,6 +122,10 @@ func (uis *UIServer) bbJiraSearch(rw http.ResponseWriter, r *http.Request) {
 		return
 	}
 	if err != nil {
+		grip.Debug(message.Fields{
```
Comment (ybrill) [inline service/ui_plugin_build_baron.go:125]: is this one necessary? It looked to me like we've already logged every way GetSearchReturnInfo could return an error.

File: service/ui_plugin_build_baron.go [hunk 5]
```diff
@@ -120,6 +122,10 @@ func (uis *UIServer) bbJiraSearch(rw http.ResponseWriter, r *http.Request) {
...
+			"message": "Chaya, service/ui_plugin_build_baron.go, 131",
```
Comment (ybrill) [inline service/ui_plugin_build_baron.go:126]: did you miss line 131 -> 126?

**Review by ybrill — CHANGES_REQUESTED** [review]

File: service/ui_plugin_build_baron.go (hunk 4 above)
Comment (malikchaya2) [reply service/ui_plugin_build_baron.go:125]: true, I'll remove it

File: graphql/ui_plugin_build_baron.go [hunk 6]
```diff
@@ -127,6 +128,11 @@ func GetSearchReturnInfo(taskId string, exec string) (*thirdparty.SearchReturnIn
 
 	if !ok {
 		projectNotFoundError = true
+		grip.Debug(message.Fields{
```
Comment (ybrill) [inline graphql/ui_plugin_build_baron.go:131]: should we also log the task id?

Comment (malikchaya2) [reply graphql/ui_plugin_build_baron.go:131]: good call!

File: service/ui_plugin_build_baron.go (hunk 5 above)
Comment (malikchaya2) [reply service/ui_plugin_build_baron.go]: removed this

**Review by ybrill — APPROVED** [review]

**Pull request merged**